src/analysis/timeline/timelineCredView.js

```diff
@@ -0,0 +1,128 @@
+// @flow
+
+import {sum} from "d3-array";
+import sortBy from "lodash.sortby";
+import stringify from "json-stable-stringify";
+import * as NullUtil from "../../util/null";
+import {TimelineCred} from "./timelineCred";
+import {type Interval, weekIntervals} from "./interval";
+import {type Node, type NodeAddressT, NodeAddress} from "../../core/graph";
+
+/**
+ * A Graph Node wrapped with cred information.
+ */
+export type CredNode = {|
+  // The Graph Node in question
+  +node: Node,
+  // The total aggregated cred. (Summed across every interval).
+  +total: number,
+  // The timeline sequence of cred (one score per interval).
+  +cred: $ReadOnlyArray<number>,
+|};
+
+/**
+ * TimelineCredView is a subset view of a TimelineCred for a user
```

Comment:
I think this docstring is inaccurate. Describing it as a "Subset view" of TimelineCred made more sense when TimelineCred still had the `.credNodes` API, etc. But now that Timeline Cred doesn't give that info, this isn't a subset view. Also, "TimelineCredView will extend this interval to be contained in the base intervals" is quite confusing.

Rather, I would say something like:

```
TimelineCredView allows a retrieving information about nodes' cred for a specific interval of time. The TimelineCredView wraps a TimelineCred instance, and provides methods like `credNode` and `credSortedNodes` for retrieving information on the available nodes and how much cred they earned. 

Every TimelineCredView is scoped to a certain view interval, which defines the min and max time of cred being viewed. When `CredNode` results are returned, the `total` field will correspond to the view interval of the TimelineCredView. By default, the view interval is chosen to include every underlying interval in the TimelineCred`.
```